affordable-housing/CSVToHTMLConverter.py

```diff
@@ -0,0 +1,191 @@
+import sys
+import csv
+import re
+import usaddress
+import urllib2
+import xml.dom.minidom
+import json
+from string import Template
+
+# @author: ashwin
+
+def stTypeToAbbrev(streetType):
+    if streetType == 'Avenue':
+        return 'AV'
+    elif streetType == 'Boulevard':
+        return 'BL'
+    elif streetType == 'Circle':
+        return 'CL'
+    elif streetType == 'Court':
+        return 'CT'
+    elif streetType == 'Drive':
+        return 'DR'
+    elif streetType == 'Expressway':
+        return 'EX'
+    elif streetType == 'Freeway':
+        return 'FY'
+    elif streetType == 'Highway':
+        return 'HY'
+    elif streetType == 'Lane':
+        return 'LN'
+    elif streetType == 'Loop':
+        return 'LP'
+    elif streetType == 'Place':
+        return 'PL'
+    elif streetType == 'Parkway':
+        return 'PY'
+    elif streetType == 'Road':
+        return 'RD'
+    elif streetType == 'Square':
+        return 'SQ'
+    elif streetType == 'Street':
+        return 'ST'
+    elif streetType == 'Terrace':
+        return 'TR'
+    elif streetType == 'Walkway':
+        return 'WW'
+    elif streetType == 'Way':
+        return 'WY'
+    else:
+        return streetType
+    
+def geocode(address, sensor=False):    
+  mapsUrl = 'https://maps.googleapis.com/maps/api/geocode/json?address='
+  url = ''.join([mapsUrl, urllib2.quote(address), '&sensor=', str(sensor).lower()])
+  jsonOutput = str(urllib2.urlopen(url).read ())  # get the response 
+  jsonOutput = jsonOutput.replace ("\\n", "")
+  result = json.loads(jsonOutput)  # converts jsonOutput into a dictionary 
+  if result['status'] != "OK": 
+    return ""
+  coordinates = result['results'][0]['geometry']['location']  # extract the geometry 
+  return str(coordinates['lng']) + ',' + str(coordinates['lat'])    
+    
+
+def mapsUrlToHref(url, addressString):
+    retval = "<a href=\"" + url + ", San Jose, CA\">" + addressString + "</a>"
+    return retval
+
+
+def templateGenerator(cells):
+    row = '<tr>'
+    for cell in cells:
+        s = Template('<td>$column</td>')
+        row = row + (s.substitute(column = cell))
+    row = row + '</tr>'
+    return row
+
+def codeEnforcerLinkGenerator(streetNum, street, streetType):
+    s = Template("<a href=javascript:loadDoc(\'$num\',\'$name\',\'$type\')>Link</a>")
+    return s.substitute(num = streetNum, name = street, type = streetType)
+
+def loadHTMLHeader(htmlfile):
```

Comment:
Instead of creating a file for the header and another for the footer, you should create one file with both with a template variable for the contents in between. This would be a good use of string.Template.